packages/backend/server/src/plugins/oauth/controller.ts

```diff
@@ -61,10 +63,14 @@ export class OAuthController {
     const state = await this.oauth.saveOAuthState({
       provider: providerName,
       redirectUri,
+      client,
+      clientNonce,
     });
 
     return {
-      url: provider.getAuthUrl(state),
+      url: provider.getAuthUrl(
+        JSON.stringify({ state, client, provider: unknownProviderName })
```

Comment:
it will breaks app before 0.21

Reply:
The old version of the app executed the code on the `app.affine.pro` online page first, shouldn't it be the latest?